m4opt/_cli/depth_map.py

```diff
@@ -0,0 +1,124 @@
+    mission: missions.Mission = getattr(missions, table.meta["args"]["mission"])
+    hpx = HEALPix(table.meta["args"]["nside"], frame=ICRS(), order="nested")
+    bandpass = table.meta["args"]["bandpass"]
+    snr = table.meta["args"]["snr"]
+    footprints = footprint_healpix(
+        hpx, mission.fov, table["target_coord"], table["roll"]
+    )
+
+    # Midpoint time of each observation
+    times = table["start_time"] + 0.5 * table["duration"]
+
+    # Evaluate the limiting magnitude for each observation, for each pixel.
+    # Note that because each field may contain a different number of pixels,
+    # we can't use Numpy array broadcasting. Instead, we repeat the entries
+    # from the observer_location and time columns an appropriate number of
+    # times.
+    #
+    # FIXME: This is a more complicated than it should be because neither
+    # `astropy.coordinates.EarthLocation` nor `astropy.time.Time` support
+    # `np.full` or `np.concatenate`.
+    with observing(
+        observer_location=EarthLocation.from_geocentric(
+            *(
+                np.concatenate(
+                    [
+                        np.tile(value, (len(footprint), 1))
+                        for value, footprint in zip(
+                            np.column_stack(
+                                [
+                                    coord.value
+                                    for coord in table["observer_location"].geocentric
+                                ]
+                            ),
+                            footprints,
+                        )
+                    ]
+                ).T
+                * table["observer_location"].geocentric[0].unit
+            )
+        ),
+        target_coord=hpx.healpix_to_skycoord(np.concatenate(footprints)),
+        obstime=times[0]
+        + np.concatenate(
+            [
+                np.full(len(footprint), value)
+                for value, footprint in zip(
+                    (times - times[0]).to_value(u.s), footprints
+                )
+            ]
+        )
+        * u.s,
+    ):
+        limmag = mission.detector.get_limmag(
+            snr,
+            np.concatenate(
+                [
+                    np.full(len(footprint), value)
+                    for value, footprint in zip(table["duration"].value, footprints)
+                ]
+            )
+            * table["duration"].unit,
+            synphot.SourceSpectrum(synphot.ConstFlux1D, amplitude=0 * u.ABmag)
+            * DustExtinction(),
+            bandpass,
+        ).to_value(u.mag)
+
+    # Find deepest limiting magnitude for each pixel
+    table = (
+        QTable(
+            {
+                "ipix": np.concatenate(
+                    (np.arange(hpx.npix), np.concatenate(footprints))
+                ),
+                "limmag": np.concatenate((np.full(hpx.npix, -np.inf), limmag)),
+            }
+        )
+        .group_by("ipix")
+        .groups.aggregate(np.max)
```

Comment:
I wonder if np.max can be the default but also make np.min and np.mean as other options in case folks want to measure like "average", "best case" or "worst case"

Reply:
This is dealing with pixels that are contained in two or more overlapping fields.